meson.build

```diff
@@ -0,0 +1,1629 @@
+    'stream/stream_slice.c',
+
+    ## Subtitles
+    'sub/ass_mp.c',
+    'sub/dec_sub.c',
+    'sub/draw_bmp.c',
+    'sub/filter_sdh.c',
+    'sub/img_convert.c',
+    'sub/lavc_conv.c',
+    'sub/osd.c',
+    'sub/osd_libass.c',
+    'sub/sd_ass.c',
+    'sub/sd_lavc.c',
+
+    ## Video
+    'video/csputils.c',
+    'video/decode/vd_lavc.c',
+    'video/filter/refqueue.c',
+    'video/filter/vf_format.c',
+    'video/filter/vf_sub.c',
+    'video/fmt-conversion.c',
+    'video/hwdec.c',
+    'video/image_loader.c',
+    'video/image_writer.c',
+    'video/img_format.c',
+    'video/mp_image.c',
+    'video/mp_image_pool.c',
+    'video/out/aspect.c',
+    'video/out/bitmap_packer.c',
+    'video/out/dither.c',
+    'video/out/dr_helper.c',
+    'video/out/filter_kernels.c',
+    'video/out/gpu/context.c',
+    'video/out/gpu/error_diffusion.c',
+    'video/out/gpu/hwdec.c',
+    'video/out/gpu/lcms.c',
+    'video/out/gpu/libmpv_gpu.c',
+    'video/out/gpu/osd.c',
+    'video/out/gpu/ra.c',
+    'video/out/gpu/shader_cache.c',
+    'video/out/gpu/spirv.c',
+    'video/out/gpu/user_shaders.c',
+    'video/out/gpu/utils.c',
+    'video/out/gpu/video.c',
+    'video/out/gpu/video_shaders.c',
+    'video/out/libmpv_sw.c',
+    'video/out/vo.c',
+    'video/out/vo_gpu.c',
+    'video/out/vo_image.c',
+    'video/out/vo_lavc.c',
+    'video/out/vo_libmpv.c',
+    'video/out/vo_null.c',
+    'video/out/vo_tct.c',
+    'video/out/win_state.c',
+    'video/repack.c',
+    'video/sws_utils.c',
+
+    ## osdep
+    'osdep/io.c',
+    'osdep/threads.c',
+    'osdep/timer.c',
+
+    'osdep/semaphore_osx.c',
+    'osdep/subprocess.c',
+
+    ## tree_allocator
+    'ta/ta.c',
+    'ta/ta_talloc.c',
+    'ta/ta_utils.c'
+)
+
+
+# osdep
+if posix
+    sources += ['input/ipc-unix.c',
+                'osdep/main-fn-unix.c',
+                'osdep/path-unix.c',
+                'osdep/polldev.c',
+                'osdep/subprocess-posix.c',
+                'osdep/terminal-unix.c',
+                'sub/filter_regex.c']
+endif
+
+if posix and host_machine.system() != 'darwin'
+    sources += 'osdep/timer-linux.c'
+endif
+
+if host_machine.system() == 'darwin'
+    sources += 'osdep/timer-darwin.c'
+endif
+
+cd_devices = {
+    'windows': 'D:',
+    'cygwin': 'D:',
+    'darwin': '/dev/disk1',
+    'freebsd': '/dev/cd0',
+    'openbsd': '/dev/rcd0c',
+    'linux': '/dev/sr0',
+}
+if host_machine.system() in cd_devices
+    cd_device = cd_devices[host_machine.system()]
+else
+    cd_device = '/dev/cdrom'
+endif
+
+dvd_devices = {
+    'windows': 'D:',
+    'cygwin': 'D:',
+    'darwin': '/dev/diskN',
+    'freebsd': '/dev/cd0',
+    'openbsd': '/dev/rcd0c',
+    'linux': '/dev/sr0',
+}
+if host_machine.system() in cd_devices
+    dvd_device = dvd_devices[host_machine.system()]
+else
+    dvd_device = '/dev/dvd'
+endif
+
+android = {
+    'name': 'android',
+    'use': host_machine.system() == 'android' and not get_option('android').disabled(),
```

Comment:
I'm curious: why do you need an option for this? When will =Dandroid be enabled, but the host machine is not android, or vice versa?